akka-cluster/src/main/resources/reference.conf

```diff
@@ -280,6 +280,35 @@ akka {
       verbose-gossip-logging = off
     }
 
+    configuration-compatibility-check {
+
+      # Enforce configuration compatibility checks when joining a cluster.
+      # Set to off to allow joining nodes to join a cluster even when configuration incompatibilities are detected or
+      # when the cluster does not support this feature. Compatibility checks are always performed and warning and
+      # error messsages are logged.
+      #
+      # This is particularly useful for rolling updates on clusters that do not support that feature. Since the old
+      # cluster won't be able to send the compatibility confirmation to the joining node, the joining node won't be able
+      # to 'know' if its allowed to join.
+      enforce-on-join = on
+
+      checkers {
+        akka-cluster = "akka.cluster.JoinConfigCompatCheckAkkaCluster"
+      }
+
+      # Some configuration properties might not be appropriate to transfer between nodes
+      # and such properties can be excluded from the the configuration compatibility check by adding
+      # the paths of the properties to this list. All properties starting with
+      # the paths defined here are excluded, i.e. you can add the path of a whole
+      # section here to skip everything inside that section.
+      sensitive-config-paths = [
+        "user.home", "user.name", "user.dir",
+        "socksNonProxyHosts", "http.nonProxyHosts", "ftp.nonProxyHosts",
+        "akka.remote.secure-cookie",
+        "akka.remote.netty.ssl.security"
+      ]
```

Comment:
I got one extra thought about this. 
Maybe we should hardcode these sensitive paths and make sure they are always excluded. Users could define here their own sensitive paths if they want and their list is can be then appended to the hardcoded one. 
WDYT?

Reply:
or we could have two properties: `sensitive-config-paths` and `library-sensitive-config-paths`.
`sensitive-config-paths` can be empty in reference.conf and defined by users in application.conf
`library-sensitive-config-paths` can be contributed to in reference.conf files (from different libraries) by using `+=`.

We have something like that for `library-extensions`.

```
library-extensions = ${?akka.library-extensions} []
```

Reply:
My concern was more about people overriding the values and leaking sensitive info. Hence the idea of some hardcoded sensitive properties.